hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeManager.java

```diff
@@ -1189,16 +1190,25 @@ public void registerDatanode(DatanodeRegistration nodeReg)
             nodes with its data cleared (or user can just remove the StorageID
             value in "VERSION" file under the data directory of the datanode,
             but this is might not work if VERSION file format has changed 
-         */        
+         */
+          // Check if nodeS's host information is same as nodeReg's, if not,
+          // it needs to update host2DatanodeMap accordringly.
+          updateHost2DatanodeMap = !nodeS.getXferAddr().equals(nodeReg.getXferAddr());
+
           NameNode.stateChangeLog.info("BLOCK* registerDatanode: " + nodeS
               + " is replaced by " + nodeReg + " with the same storageID "
-              + nodeReg.getDatanodeUuid());
+              + nodeReg.getDatanodeUuid() + ", updateHost2DatanodeMap: " + updateHost2DatanodeMap);
```

Comment:
Is this extra information needed at the `INFO` level log? I understand that having the value printed is helpful during development, but I don't think it's meaningful to an operator.

Also, if you're here to change a log message, maybe also change it to use the format string version instead of string concatenation?

Reply:
Yeah, agree. Let me undo this change.

Reply:
I am going to upload the patch which does not log updateHost2DatanodeMap.